src/Microsoft.AspNetCore.Sockets.Http/HttpConnectionDispatcher.cs

```diff
@@ -363,6 +364,9 @@ private Task ProcessNegotiate(HttpContext context, HttpSocketOptions options, Co
             // Establish the connection
             var connection = _manager.CreateConnection();
 
+            //Set the IHttpConnectionFeature now that we can access it.
+            connection.Features.Set(context.Features.Get<IHttpConnectionFeature>());
```

Comment:
Why do it here? User code never accessing the connection at this point right?

Reply:
I put it here because it was the earliest place we could. Where do you think would be a better spot?

Reply:
I think it isn't required here at all.

Reply:
It is. The feature isn't set otherwise and we get null refs in the tests.

Reply:
Are the tests broken? I think we should be able to move this to EnsureConnectionState

Reply:
Yeah, the issue was human error. Working now 😄